docs/using-the-jdbc-driver/using-plugins/UsingTheHostMonitoringPlugin.md

```diff
@@ -72,3 +72,17 @@ properties.setProperty("monitoring-socketTimeout", "10");
 > We recommend you either disable the Host Monitoring Connection Plugin or avoid using RDS Proxy endpoints when the Host Monitoring Connection Plugin is active.
 >
 > Although using RDS Proxy endpoints with the AWS Advanced JDBC Driver with Enhanced Failure Monitoring doesn't cause any critical issues, we don't recommend this approach. The main reason is that RDS Proxy transparently re-routes requests to a single database instance. RDS Proxy decides which database instance is used based on many criteria (on a per-request basis). Switching between different instances makes the Host Monitoring Connection Plugin useless in terms of instance health monitoring because the plugin will be unable to identify which instance it's connected to, and which one it's monitoring. This could result in false positive failure detections. At the same time, the plugin will still proactively monitor network connectivity to RDS Proxy endpoints and report outages back to a user application if they occur.
+
+# Host Monitoring Plugin v2
```

Comment:
```
# **Experimental** Host Monitoring Plugin v2

> [!WARNING] This plugin is experimental and should be verified before using it in production environment.

This plugin is an alternative implementation of the enhanced failure monitoring feature and it is functionally equivalent to the Host Monitoring Plugin described above. Both plugins share the same set of [configuration parameters](#enhanced-failure-monitoring-parameters). The `efm2` plugin is designed to be a drop-in replacement for the `efm` plugin.

The `efm2` plugin is designed to address [some of the issues](https://github.com/awslabs/aws-advanced-jdbc-wrapper/issues/675) reported by multiple users. The following changes have been made:
- use weak pointers to ease garbage collection
- split monitoring logic into two separate threads to increase overall monitoring stability
- review locks for monitoring context
- redesign stopping of idle monitoring threads
- simplify monitoring logic

> [!NOTE] Since `efm` and `efm2` are separate plugins, it is technically possible to specify both plugins for a single connection. While this should not have any negative side effects, it is not recommended. It is recommended to use either the `efm` plugin, or the `efm2` plugin where it's needed.
```